### ISSUE TITLE
Feature request: support `-typed` flag in `gomock` rule

### ISSUE THREAD

**Issue by maratori** [bazelbuild/rules_go#4012]
### What version of rules_go are you using?

0.47.0

### What version of gazelle are you using?

0.36.0

### What version of Bazel are you using?

Bazelisk version: v1.19.0
Build label: 7.2.1
Build target: @@//src/main/java/com/google/devtools/build/lib/bazel:BazelServer
Build time: Tue Jun 25 15:53:57 2024 (1719330837)
Build timestamp: 1719330837
Build timestamp as int: 1719330837

### Does this issue reproduce with the latest releases of all the above?

Yes

### Feature request

https://github.com/golang/mock is not maintained anymore, and everyone uses https://github.com/uber-go/mock fork. It has a new flag:
```
-typed: Generate Type-safe 'Return', 'Do', 'DoAndReturn' function. (default false)
```

It would be great it the `gomock` rule supported this flag.
